src/main/java/org/generationcp/middleware/api/location/LocationServiceImpl.java

```diff
@@ -61,6 +64,27 @@ public long countFilteredLocations(final LocationSearchRequest locationSearchReq
 	@Override
 	public void deleteLocation(final Integer locationId) {
 		final Location location = this.daoFactory.getLocationDAO().getById(locationId);
+		List<Locdes> Blocks = null;
+		if (location.getLtype() == 415) { // Field
```

Comment:
@cuenyad create (or find an existing place) an enum (or use constants) for 415 and 416, and also use it in BMSAPI because the values are also used there

Reply:
done.